src/common/registry-csm/main/key-registry/key-registry.module.ts

```diff
@@ -0,0 +1,56 @@
+import { DynamicModule, Module } from '@nestjs/common';
+import { RegistryModuleSyncOptions, RegistryModuleAsyncOptions } from '../interfaces/module.interface';
+import { CSMKeyRegistryService } from './key-registry.service';
+import { RegistryStorageModule } from '../../../registry/storage/registry-storage.module';
+import { RegistryFetchModule } from '../../fetch/registry-fetch.module';
+import { REGISTRY_GLOBAL_OPTIONS_TOKEN } from '../constants';
+
+@Module({
+  imports: [RegistryStorageModule],
+  providers: [CSMKeyRegistryService],
+  exports: [CSMKeyRegistryService, RegistryStorageModule],
+})
+export class CSMKeyRegistryModule {
+  static forRoot(options?: RegistryModuleSyncOptions): DynamicModule {
+    return {
+      global: true,
+      ...this.forFeature(options),
+    };
+  }
+
+  static forRootAsync(options: RegistryModuleAsyncOptions): DynamicModule {
+    return {
+      global: true,
+      ...this.forFeatureAsync(options),
+    };
+  }
+
+  static forFeature(options?: RegistryModuleSyncOptions): DynamicModule {
+    return {
+      module: CSMKeyRegistryModule,
+      imports: [...(options?.imports || []), RegistryFetchModule.forFeature(options)],
+      providers: [
+        {
+          provide: REGISTRY_GLOBAL_OPTIONS_TOKEN,
```

Comment:
You are right that this can be a problem, but in our case we are using Symbol and Symbol has the property that Symbol(‘registryGlobalOptions’) !== Symbol(‘registryGlobalOptions’)